### PR TITLE
hplip: 3.23.3 -> 3.23.8

### PR DESCRIPTION
**Pull request by arthsmn**
## Description of changes

https://developers.hp.com/hp-linux-imaging-and-printing/release_notes

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by IogaMaster — APPROVED** [review]
LGTM

**Comment by IogaMaster** [comment]
@ofborg eval hplip

File: pkgs/misc/drivers/hplip/default.nix [hunk 1]
```diff
@@ -14,16 +14,16 @@
 let
 
   pname = "hplip";
-  version = "3.23.3";
+  version = "3.23.8";
 
   src = fetchurl {
     url = "mirror://sourceforge/hplip/${pname}-${version}.tar.gz";
-    sha256 = "sha256-5CYKmKKx2I0h6CVi3kGaohyVvJ4qzjWDNGqA/SF+B7Y=";
+    sha256 = "sha256-98wF9ijAz9dQ5UrkFDHB390p6XaC8YtcW6XLLFtLG0Y=";
```
Comment (eclairevoyant) [inline pkgs/misc/drivers/hplip/default.nix:21]:
```suggestion
    hash = "sha256-98wF9ijAz9dQ5UrkFDHB390p6XaC8YtcW6XLLFtLG0Y=";
```

File: pkgs/misc/drivers/hplip/default.nix [hunk 2]
```diff
@@ -14,16 +14,16 @@
...
   };
 
   plugin = fetchurl {
     url = "https://developers.hp.com/sites/default/files/${pname}-${version}-plugin.run";
-    sha256 = "sha256-AyZBiF1B42dGnJeoJLFSCGNK83c86ZAM2uFciuv2H4A=";
+    sha256 = "sha256-frsgye3f0M3HE2trKRlfFvMnDEwe+z74IumCdVPrcSY=";
```
Comment (eclairevoyant) [inline pkgs/misc/drivers/hplip/default.nix:26]:
```suggestion
    hash = "sha256-frsgye3f0M3HE2trKRlfFvMnDEwe+z74IumCdVPrcSY=";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
I don't have a way of testing this, but one point of feedback:

**Comment by IogaMaster** [comment]
I wonder why this package is failing, because it works for me, as of right now.

**Comment by eclairevoyant** [comment]
FYI please squash the commits so that it follows contributing.md
Otherwise looks fine

**Comment by IogaMaster** [comment]
Where did these come from:
image

**Comment by IogaMaster** [comment]
I think you are working from master and updated instead of using a separate branch.

**Comment by arthsmn** [comment]
> I think you are working from master and updated instead of using a separate branch.

Sorry, I was doing a test on another package so I messed up lol

**Review by IogaMaster — APPROVED** [review]
LGTM

Package is functional.

**Review by Janik-Haag — COMMENTED** [review]
Update looks good.
hp-check says there is a bunch of missing dependencies, I'm fine merging if this is the expected behavior.

**Comment by arthsmn** [comment]
> Update looks good. hp-check says there is a bunch of missing dependencies, I'm fine merging if this is the expected behavior.

This is just an update. This package has problems (my printer isn't working properly, trying to figure out why) but nothing that would affect existing users, as the new package seems to just support new printers and distributions.

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**